modules/dns/dns_api.rb

```diff
@@ -73,12 +78,34 @@ class Api < ::Sinatra::Base
       end
     end
 
-    def validate_dns_name!(name)
-      raise Proxy::Dns::Error.new("Invalid DNS name #{name}") unless name =~ /^([a-zA-Z0-9]([-a-zA-Z0-9]+)?\.?)+$/
+    def validate_srv_value!(value)
+      parts = value.split(' ')
+      priority,weight,port,target = parts
+      validate_dns_name!(target, 'SRV')
+      unless parts.size == 4 &&
+        (0..65535) === Integer(priority) &&
+        (0..65535) === Integer(weight) &&
+        (1..65535) === Integer(port)
+          raise
+      end
+    rescue ArgumentError, RuntimeError # if any of the three tests on integer above fails or the condition was not met
+      raise Proxy::Dns::Error.new("Invalid DNS SRV value #{value}")
+    end
+
+    def validate_srv_name!(name)
+      raise Proxy::Dns::Error.new("Invalid DNS srv name #{name}") unless name =~ /^([a-zA-Z0-9_]([-a-zA-Z0-9_]+)?\.?)+$/
+    end
+
+    def validate_dns_name!(name, type)
```

Comment:
The function remains compatible if you use `type=nil`. Not sure if plugins use this function.